app/models/host.rb

```diff
@@ -656,12 +659,17 @@ def verify_credentials(auth_type = nil, options = {})
     when 'ws'     then verify_credentials_with_ws(auth_type)
     when 'ipmi'   then verify_credentials_with_ipmi(auth_type)
     else
-      verify_credentials_with_ws(auth_type)
+      verify_credentials_default(auth_type, options)
     end
 
     true
   end
 
+  # some providers (i.e.: openstack) want to use something other than ws as a default
+  def verify_credentials_default(auth_type, _options)
+    verify_credentials_with_ws(auth_type)
+  end
+
```

Comment:
Would it be fair to say the inverse, that vmware wants to use ws as the default?

Also am I reading this wrong or does openstack "default" to ws as well?
```
    case auth_type.to_s
    when 'remote', 'default', 'ssh_keypair' then verify_credentials_with_ssh(auth_type, options)
    when 'ws'                               then verify_credentials_with_ws(auth_type)
    when 'ipmi'                             then verify_credentials_with_ipmi(auth_type)
    else
      verify_credentials_with_ws(auth_type)
    end
```

Reply:
no. openstack does not implement ws.
so while I agree with you that they fall through to ws, that means they will throw an error.

If you also note, they added 'default' and ssh_keypair to go to ssh.

So I actually read they default to ssh, but they just left the else code because they didn't understand the implications.

In the end, I am extracting credentials outside of this method. Too many methods override this. I no longer have to make this change, but it is still weighing on my mind.

Reply:
I do wonder what the default should be for the base class (and openstack).
Still think HostEsx should default to ws, but think the others should be ssh

Reply:
you read this correctly.

vmware EsxHost is the only one that wants ws as the default
but we have it setup that ws is the default for all

Most of the providers have overwritten verify_credentials so I'm unsure if this method helps us that much.

Reply:
@agrare do you want to set `ssh` to the default?
We can update VMware EsxHost to use `ws` as the default.

Reply:
:+1: yes if VMware is the only different one seems better to flip the default